### PR TITLE
Remove race conditions from run_lock_spec

### PR DESCRIPTION
**Pull request by jkeiser**

### CODE REVIEW

File: spec/functional/run_lock_spec.rb [hunk 1]
```diff
@@ -427,6 +433,8 @@ def create_lock
       end
     end
 
+    attr_writer :run_lock
```
Comment (danielsdeleo) [inline spec/functional/run_lock_spec.rb]: What's this for? I don't see `self.run_lock = thing` elsewhere in the diff, or see any `@run_lock` references.

**Comment by danielsdeleo** [comment]
One line looks like maybe a leftover from debugging, 👍 on the rest

File: spec/functional/run_lock_spec.rb (hunk 1 above)
Comment (jkeiser) [inline spec/functional/run_lock_spec.rb]: Cruft! I'll remove.

**Comment by lamont-granquist** [comment]
:+1:

**Pull request merged**